dipy/segment/clustering.py

```diff
@@ -657,3 +658,63 @@ def _traverse(node, level=0):
 
         _traverse(self.root)
         return clusters
+
+
+def qbx_with_merge(streamlines, thresholds,
+                   nb_pts=20, select_randomly=None, verbose=True):
+    if verbose:
+        t = time()
+    len_s = len(streamlines)
+    if select_randomly is None:
+        select_randomly = len_s
+    indices = np.random.choice(len_s, min(select_randomly, len_s),
+                               replace=False)
+    sample_streamlines = set_number_of_points(streamlines, nb_pts)
+
+    if verbose:
+        print(' Resampled to {} points'.format(nb_pts))
+        print(' Size is %0.3f MB' % (nbytes(sample_streamlines),))
+        print(' Duration of resampling is %0.3f sec.' % (time() - t,))
+        print(' QBX phase starting...')
+
+    qbx = QuickBundlesX(thresholds,
+                        metric=AveragePointwiseEuclideanMetric())
+
+    if verbose:
+        t1 = time()
+    qbx_clusters = qbx.cluster(sample_streamlines, ordering=indices)
+
+    if verbose:
+        print(' Merging phase starting ...')
+
+    qbx_merge = QuickBundlesX([thresholds[-1]],
+                              metric=AveragePointwiseEuclideanMetric())
+
+    final_level = len(thresholds)
+
+    qbx_ordering_final = np.random.choice(
```

Comment:
You could reuse the `rng` from above here instead of relying on the global RandomState.